avaframe/ana3AIMEC/ana3AIMEC.py

```diff
@@ -765,68 +784,296 @@ def analyzeFields(rasterTransfo, pLim, newRasters, cfgPath):
+    resAnalysis['TP'] = TP
+    resAnalysis['FN'] = FN
+    resAnalysis['FP'] = FP
+    resAnalysis['TN'] = TN
+
+    return resAnalysis
+
+
 
+def analyzeMass(fnameMass):
+    """ Analyse Mass data
+
+    Parameters
+    ----------
+    fnameMass: list
+        list of path to mass data to analyse
+
+    Returns
+    -------
+    relMass: 1D numpy array
+        containing for each simulation analyzed the
+        release mass
+    entMass: 1D numpy array
+        containing for each simulation analyzed the
+        entrained mass
+    finalMass: 1D numpy array
+        containing for each simulation analyzed the
+        final mass
+    relativMassDiff: 1D numpy array
+        containing for each simulation analyzed
+        the final mass diff with ref (in %)
+    growthIndex: 1D numpy array
+        containing for each simulation analyzed the
+        growth index
+    growthGrad: 1D numpy array
+        containing for each simulation analyzed the
+        growth gradient
+    """
     # initialize Arrays
-    nTopo = len(fname)
+    nTopo = len(fnameMass)
     massDiffers = False
-    runout = np.empty((3, nTopo))
-    runoutMean = np.empty((3, nTopo))
-    ampp = np.empty((nTopo))
-    mmpp = np.empty((nTopo))
-    amd = np.empty((nTopo))
-    mmd = np.empty((nTopo))
-    ams = np.empty((nTopo))
-    mms = np.empty((nTopo))
-    elevRel = np.empty((nTopo))
-    deltaH = np.empty((nTopo))
     grIndex = np.empty((nTopo))
     grGrad = np.empty((nTopo))
     releaseMass = np.empty((nTopo))
     entrainedMass = np.empty((nTopo))
     finalMass = np.empty((nTopo))
     relativMassDiff = np.empty((nTopo))
+    time = 0
 
-    n = np.shape(lcoord)[0]
-    pCrossAll = np.zeros((nTopo, len(scoord)))
+    log.info('Analyzing mass')
+    log.info('{: <10} {: <10} {: <10}'.format('Sim number ', 'GI ', 'GR '))
     # For each data set
     for i in range(nTopo):
-        rasterdataPres = dataPressure[i]
-        rasterdataDepth = dataDepth[i]
-        rasterdataSpeed = dataSpeed[i]
-        rasterArea = rasterTransfo['rasterArea']
-        # rasterArea[np.where(np.isnan(rasterdataPres))] = np.nan
+        # analyze mass
+        releaseMass[i], entrainedMass[i], finalMass[i], grIndex[i], grGrad[i], entMass, totalMass, time = readWrite(
+            fnameMass[i], time)
+        if i == 0:
+            entMassArray = np.zeros((nTopo, np.size(time)))
+            totalMassArray = np.zeros((nTopo, np.size(time)))
+        entMassArray[i] = entMass
+        totalMassArray[i] = totalMass
+        relativMassDiff[i] = (finalMass[i]-finalMass[0])/finalMass[0]*100
+        if not (releaseMass[i] == releaseMass[0]):
+            massDiffers = True
+        log.info('{: <10} {:<10.4f} {:<10.4f}'.format(*[i+1, grIndex[i], grGrad[i]]))
+    if massDiffers:
+        log.warning('Release masses differs between simulations!')
+
+    return releaseMass, entrainedMass, entMassArray, totalMassArray, finalMass, relativMassDiff, grIndex, grGrad, time
```

Comment:
This is a lot of variables to return, could we combine the mass* ones into one?

Reply:
made an issue